idea_town/settings.py

```diff
@@ -0,0 +1,167 @@
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+]
+
+for app in config('EXTRA_APPS', default='', cast=Csv()):
+    INSTALLED_APPS.append(app)
+
+
+MIDDLEWARE_CLASSES = (
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    'csp.middleware.CSPMiddleware',
+)
+
+ROOT_URLCONF = 'idea_town.urls'
+
+WSGI_APPLICATION = 'idea_town.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/1.7/ref/settings/#databases
+
+DATABASES = {
+    'default': config(
+        'DATABASE_URL',
+        cast=dj_database_url.parse
+    )
+}
+
+# Internationalization
+# https://docs.djangoproject.com/en/1.7/topics/i18n/
+
+LANGUAGE_CODE = config('LANGUAGE_CODE', default='en-us')
+
+TIME_ZONE = config('TIME_ZONE', default='UTC')
+
+USE_I18N = config('USE_I18N', default=True, cast=bool)
+
+USE_L10N = config('USE_L10N', default=True, cast=bool)
+
+USE_TZ = config('USE_TZ', default=True, cast=bool)
+
+STATIC_ROOT = config('STATIC_ROOT', default=os.path.join(BASE_DIR, 'static'))
+STATIC_URL = config('STATIC_URL', '/static/')
+STATICFILES_STORAGE = 'whitenoise.django.GzipManifestStaticFilesStorage'
+
+MEDIA_ROOT = config('MEDIA_ROOT', default=os.path.join(BASE_DIR, 'media'))
+MEDIA_URL = config('MEDIA_URL', '/media/')
+
+SESSION_COOKIE_SECURE = config('SESSION_COOKIE_SECURE', default=not DEBUG, cast=bool)
+
+TEMPLATES = [
+    {
+        'BACKEND': 'django_jinja.backend.Jinja2',
+        'APP_DIRS': True,
+        'OPTIONS': {
+            'match_regex': r'^(?!!(admin|registration)/.*)',
+            'match_extension': '.html',
+            'newstyle_gettext': True,
+            'context_processors': [
+                'idea_town.base.context_processors.settings',
+                'idea_town.base.context_processors.i18n',
+            ],
+        }
+    },
+    {
+        'BACKEND': 'django.template.backends.django.DjangoTemplates',
+        'APP_DIRS': True,
+        'OPTIONS': {
+            'context_processors': [
+                'django.contrib.auth.context_processors.auth',
+                'django.template.context_processors.debug',
+                'django.template.context_processors.i18n',
+                'django.template.context_processors.media',
+                'django.template.context_processors.static',
+                'django.template.context_processors.tz',
+                'django.contrib.messages.context_processors.messages',
+            ],
+        }
+    },
+]
+
+# Django-CSP
+CSP_DEFAULT_SRC = (
+    "'self'",
+)
+CSP_FONT_SRC = (
+    "'self'",
+    'http://*.mozilla.net',
```

Comment:
we'll probably want to also whitelist the `*.firefox.com` and `.mozaws.net` domains for all these CSP rules. I'm not sure if the `*` lets us go two subdirs deep or not (for `ideatown.dev.mozaws.net` and `ideatown.stage.mozaws.net`)